src/net_processing.cpp

```diff
@@ -3881,10 +3882,10 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
                     for (std::set<uint256>::iterator it = pto->m_tx_relay->setInventoryTxToSend.begin(); it != pto->m_tx_relay->setInventoryTxToSend.end(); it++) {
                         vInvTx.push_back(it);
                     }
-                    CAmount filterrate = 0;
+                    CFeeRate filterrate;
```

Comment:
FWIW identical code at L3847 and L3885 (also the case before the changes touching this).
```cpp
                    CFeeRate filterrate;
                    {
                        LOCK(pto->m_tx_relay->cs_feeFilter);
                        filterrate = CFeeRate(pto->m_tx_relay->minFeeFilter);
                    }
```

Reply:
The p2p code is the most daunting part. Refactoring has its own dangers. Complete ACK on working to make it easier to understand!